### PR TITLE
Docs automation

### PR DESCRIPTION
**Pull request by jonatanklosko**
Recently The Board requested to put all the Motions on the website. This PR handles this and sets up more automation regarding new documents.

I created a new repo [thewca/documents](https://github.com/thewca/wca-documents) for this purpose. Those documents could certainly stay within our `public` folder, but given that the number of documents is rapidly growing, and that they are a whole separate concern, I think it makes sense to just separate them. Let me know if you think otherwise!
*Thought: This way we may upload documents to the repo using GitHub UI, which is not the practice for the website, but would probably be just fine for the documents purpose.*

It's currently deployed to the staging and you can see how it looks [here](https://staging.worldcubeassociation.org/documents).

Remaining points:
- [x] Copy the documents when spinning up a new server (@jfly ?)
- [x] Would it make sense to remove timestamp from the Bylaws file? (so that we don't have to update the code whenever it changes). (@AlbertoPdRF ?)

### CODE REVIEW

**Comment by AlbertoPdRF** [comment]
This looks great to me! And yes, I think we can remove the timestamp from the Bylaws PDF.

**Comment by viroulep** [comment]
For copying the documents, maybe just add a git clone somewhere in the chef files?

Also the motions and bylaws are available as markdown in [this repo](https://github.com/thewca/wca-bylaws), so it might also make sense to not have a duplicate repo with the "built" version of it, but just build them on the fly (but Alberto is probably still working on producing a correct pdf out of the markdown, as there is a bunch of aesthetics to fix with the basic pandoc+wkhtmltopdf approach).

**Comment by AlbertoPdRF** [comment]
The other option may be to treat the repo wca-bylaws as we treat the wca-regulations one? I.e. not uploading PDFs for the Motions and Bylaws, but "rendering" the .md files on the website? Sorry if rendering is not the appropriate term :P

**Comment by jonatanklosko** [comment]
I didn't know that the PDF documents were build out of Markdown on a regular basis. There are certainly many ways to go, one I think of would be having all the documents in Markdown and letting Travis build PDFs out of them to a separate branch. We could then clone that branch and use on the website.

@AlbertoPdRF I thought of that, but I somehow feel that official documents should rather be something like PDFs instead of content on our website (and keeping both doesn't seem reasonable).

What do you think is best?

File: chef/site-cookbooks/wca/templates/wca.screenrc.erb [hunk 1]
```diff
@@ -7,7 +7,7 @@ screen -t run
 <% if @rails_env == "development" %>
 stuff "RACK_ENV=development DATABASE_URL=<%= @db_url %> bundle exec rails server\012"
 <% else %>
-stuff "../scripts/deploy.sh rebuild_rails\012"
+stuff "../scripts/deploy.sh rebuild_rails update_docs\012"
```
Comment (jonatanklosko) [inline chef/site-cookbooks/wca/templates/wca.screenrc.erb]: I believe this is the place responsible for the first deployment on a newly spun up server.

**Comment by jonatanklosko** [comment]
Ok, I think this is ready. After discussion on Slack with @AlbertoPdRF and @viroulep I set up [thewca/wca-documents](https://github.com/thewca/wca-documents) so that it:
- stores all WCA Documents in Markdown format
- [Travis] builds PDFs into a `build` branch whenever some changes are made
- [Travis] runs `deploy.sh update_docs` on the server, which effectively brings the newly generated PDFs over to it

While working on documents styling I naturally tested the deployment (to the staging) multiple times and it works like a charm. The build usually takes around a minute, so given that Travis needs to start, the documents are live in ~2 mins from an update to the repo, no additional actions involved :tada:

File: WcaOnRails/app/assets/stylesheets/collapsible-panels.scss [hunk 2]
```diff
@@ -15,7 +15,8 @@
  * Make collapse link fill up entire panel-title
  * Copied from http://stackoverflow.com/a/20188613
  */
-.panel-title a[data-toggle="collapse"] {
+.panel-title a[data-toggle="collapse"],
+.panel-title a.full-space {
```
Comment (viroulep) [inline WcaOnRails/app/assets/stylesheets/collapsible-panels.scss]: Why that selector change?
Isn't there any other place where we would need to add this "full-space" class?

File: chef/site-cookbooks/wca/templates/wca.screenrc.erb (hunk 1 above)
Comment (viroulep) [reply chef/site-cookbooks/wca/templates/wca.screenrc.erb]: I think so too.
The Regulations use their own recipe to be built, but it should be fine to get the docs here!

File: WcaOnRails/app/assets/stylesheets/collapsible-panels.scss (hunk 2 above)
Comment (jonatanklosko) [reply WcaOnRails/app/assets/stylesheets/collapsible-panels.scss]: This makes an `a` tag take the whole space of a panel title.
The first selector applies to `a` tags that `collapse` the given panel back and forth with some bootstrap JavaScript. I wanted to reuse the same styling for the Bylaws panel, but it doesn't have `[data-toggle="collapse"]`, it's just a normal link!

**Pull request merged**